src/OmniSharp.Host/Services/AnalyzerAssemblyLoader.cs

```diff
@@ -0,0 +1,265 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Host.Services
+{
+    // This is shadow copying loader. Makes sure that analyzer assemblies are not locked
+    // on disk during analysis.
+    public class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
```

Comment:
since this is strongly based on the Roslyn implementation, I would recommend to include the original copyright header too https://github.com/dotnet/roslyn/blob/master/src/Compilers/Shared/ShadowCopyAnalyzerAssemblyLoader.cs#L1 with a note that it's derived

Reply:
Fixed at https://github.com/OmniSharp/omnisharp-roslyn/pull/1474/commits/b33d245f018e14a66b7346d1fd6aa1dde76c33c1